api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -233,8 +253,21 @@ message CertificateValidationContext {
   //
   // See :ref:`the TLS overview <arch_overview_ssl_enabling_verification>` for a list of common
   // system CA locations.
+  //
+  // If *trusted_ca* is a filesystem path, a watch will be added to the parent
+  // directory for any file moves to support rotation. This currently only
+  // applies to dynamic secrets, when the *CertificateValidationContext* is
+  // delivered via SDS.
   config.core.v3.DataSource trusted_ca = 1;
 
+  // If specified, updates of a file-based *trusted_ca* source will be triggered
+  // by this watch. This allows explicit control over the path watched, by
+  // default the parent directory of the filesystem path in *trusted_ca* is
+  // watched if this field is not specified. This only applies when a
+  // *CertificateValidationContext* is delivered by SDS with references to
+  // filesystem paths.
+  config.core.v3.WatchedDirectory watched_directory = 11;
```

Comment:
I still don't understand how this does anything if there is only a single file above? What's the utility?

Reply:
Let's say you have a trusted CA files `/foo/bar/baz/ca.pem`. By default, envoy watches `/foo/bar/baz`. If you want to reload when `baz` is atomically moved, you need to set the watch on `/foo/bar`. This is what this field allows.

Reply:
OK I think having a concrete example might help for this stuff (or ref link back to the docs where you have the example) as it was pretty hard for me to understand the difference.